docs/sign-in-out.md

```diff
@@ -3,46 +3,117 @@ title: Sign In and Out
 layout: docs
 ---
 
-To sign in and out of Tableau Server, call the server's `.auth.signin` method in a `with` block.
+The first step to using the TSC library is to sign in to your Tableau Server (or Tableau Online). This page explains how to sign in, sign out, and switch sites, with examples for both Tableau Server and Tableau Online.
+
+* TOC
+{:toc}
+
+## Sign In
+
+Signing in can be done two different ways:
+
+* Personal Access Tokens - In most cases this is the preferred method because it improves security by avoiding the need to use or store passwords directly. Access tokens also expire by default if not used after 15 consecutive days. This option is available for Tableau Server 2019.4 and newer. Refer to [Personal Access Tokens](https://help.tableau.com/current/server/en-us/security_personal_access_tokens.htm) for more details.
+* Username and Password - Direct sign in with account username and password.
+
+Examples showing both of these cases are included below.
+
+**Note:** When you sign in, the TSC library manages the authenticated session for you. However, the validity of the underlying credentials token is limited by the maximum session length set on your Tableau Server (2 hours by default).
+
+### Sign in with Personal Access Token
+
+To sign in to Tableau Server or Tableau Online with a personal access token, you'll need the following values:
+
+Name | Description
+:--- | :---
+TOKEN_NAME | The personal access token name (from the My Account settings page)
+TOKEN_VALUE | The personal access token value (from the My Account settings page)
+**Tableau Server** |
+SITENAME | The Tableau Server site you are authenticating with; for example in the site URL http://MyServer/#/site/MarketingTeam/projects, the site name is MarketingTeam; in the REST API documentation this field is also referred to as contentUrl; this value can be omitted to connect with the Default site on the server
+SERVER_URL | The Tableau Server you are authenticating with; if your server has SSL enabled, this should be an HTTPS link
+**Tableau Online** |
+SITENAME | The Tableau Online site you are authenticating with; for example in the site URL https://10ay.online.tableau.com/#/site/umbrellacorp816664/workbooks, the site name is umbrellacorp816664; in the REST API documentation this field is also referred to as contentUrl; this value is always required when connecting to Tableau Online
+SERVER_URL | The Tableau Online instance you are authenticating with; in the example above the server URL would be https://10ay.online.tableau.com; this will always be an an HTTPS link
+
+This example illustrates using the above values to sign in with a personal access token, do some operations, and then sign out:
+
+```py
+import tableauserverclient as TSC
+
+tableau_auth = TSC.PersonalAccessTokenAuth('TOKEN_NAME', 'TOKEN_VALUE', 'SITENAME')
+server = TSC.Server('https://SERVER_URL')
```

Comment:
I think it would be helpful to add `use_server_version=True` here or `server.use_server_version()` below, to exemplify best practices

Reply:
Agreed